lms/views/dashboard/api/user.py

```diff
@@ -187,15 +192,32 @@ def students_metrics(self) -> APIStudents:
                         annotations=0, replies=0, last_activity=None
                     ),
                 )
-
-            if assignment.auto_grading_config:
-                api_student["auto_grading_grade"] = (
-                    self.auto_grading_service.calculate_grade(
-                        assignment.auto_grading_config,
-                        api_student["annotation_metrics"],
-                    )
-                )
-
             students.append(api_student)
 
+        if assignment.auto_grading_config:
+            students = self._add_auto_grading_data(assignment, students)
+
         return {"students": students}
+
+    def _add_auto_grading_data(
```

Comment:
Helper function to add the auto graind details, the main function was getting a bit out of hand.